crates/bws/install.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/sh
+set -u
+
+##################################################
+# An installer for the bws command line utility. #
+##################################################
+
+BWS_VERSION="${BWS_VERSION:-0.5.0}"
+
+main() {
+  check_required
+  platform_detect
+  arch_detect
+  download_bws
+  validate_checksum
+  install_bws
+}
+
+error() {
+  echo "$1" >&2
+  echo "Exiting..." >&2
+  exit 1
+}
+
+check_required() {
```

Comment:
I'm thinking let's potentially check to make sure some of the other programs required are present... Like `sha256sum`, etc.

Reply:
I also did not recognize some commands in the script, they also do not seem to be on my system, at least by default. Examples are `error`, `extract`, etc unless I am missing something 🙂 (Which is very possible, I am not an expert in bash scripts!)

Reply:
Ah... It looks like `sha256sum` is a part of `coreutils` on macOS, which has to be installed. I was trying to use only pre-installed executables where possible. I'll see if macOS has a preinstalled way to do this and go with that if I can.

Reply:
`error` and `extract` are defined in [here](https://github.com/bitwarden/sdk/pull/645/files#diff-24589d5888f18206c9e01fa697e2ec59167c759cbced32664241bc456166f373R19-R23) and [here](https://github.com/bitwarden/sdk/pull/645/files#diff-24589d5888f18206c9e01fa697e2ec59167c759cbced32664241bc456166f373R82-R84).

Reply:
It looks like `shasum -a 256` is the way to go for macOS. `shasum` is included in default macOS installations, so I just need to add some special handling there for macOS.

Reply:
I added a fallback for checksum validation on macOS installations that haven't installed `coreutils` [here](https://github.com/bitwarden/sdk/pull/645/files#diff-24589d5888f18206c9e01fa697e2ec59167c759cbced32664241bc456166f373R74-R80).

Reply:
Awesome, thanks!